### ISSUE TITLE
Keyboard input character error

### ISSUE THREAD

**Issue by MNRK01** [ipython/ipython#12119]
I have a strange error with the ipython console that started a few days ago with the ipython 7.12.0 update. My python environment is Anaconda 2019.10.

When I start up the Anaconda prompt, everything appears as expected. However , when I try to input text '1+2' repeatedly, the following cells appears.

```
(base) C:\Users\USER01>ipython
Python 3.7.6 (default, Jan  8 2020, 20:23:39) [MSC v.1916 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.12.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: 1+222
Out[1]: 3

In [2]: 11++222222
Out[2]: 3

In [3]: 111+++222222222
Out[3]: 3

In [4]:
```

It appears that there is some issue with characters being reflected back to the ipython console. This does not appear to be a problem with the ipython console started within Spyder. Not surprisingly, backspace functionality is also broken.

I'm not sure where the issue lies, but any help that you can provide to fix it would be appreciated.

Thanks.

**Comment by Carreau** [comment]
Can you try different versions ? 
Does reverting to 7.11 works ?

**Comment by MNRK01** [comment]
I just tried reverting back to 7.11.1 and the issue is still there. So, I think that the connection to the ipython update is coincidental. FWIW, this is also happening with jupyter console, although the output is slightly different than above when repeatedly typing '1+2'.

```
(base) C:\Users\USER01>jupyter console --kernel=python3
Jupyter console 6.1.0

Python 3.7.6 (default, Jan  8 2020, 20:23:39) [MSC v.1916 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.11.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: 1+2
Out[1]: 3

In [2]: 11++22
Out[2]: 3

In [3]: 111+++222
Out[3]: 3
```

I downgraded jupyter_console from 61.0 to 6.0.0 which also downgraded prompt_toolkit from 3.0 to 2.0.9. After this downgrade, the issue is gone, although the REPL seems slow. But, it works.  I then upgraded to ipython 7.12.0 and it works too! So the issue is coming from the jupyter_console 6.1.0 update rather than the ipython update.

I'll close this issue and move my question to the jupyter_console github site.

**Comment by MNRK01** [comment]
I figured out the issue.  I had TERM='dumb' in my environment. It was a holdover from way back. This messes up prompt_toolkit >= 3.0.3 which is causing the issue described above. Additionally, it also killed the color scheme for ipython, something that I noticed after I created this PR.

I looked at the diffs from v. 3.0.2 and v. 3.0.3 of prompt_toolkit and I think that https://github.com/prompt-toolkit/python-prompt-toolkit/commit/61a6eb6747e5eff1c3a8d5394176c3d9735c7f99 is the one that did it. It has a change that sets cls.DEPTH_1_BIT if TERM=="dumb".

Once I unset the TERM environment variable, everything works well!
